src/com/mjcode/model/Tile.java

```diff
@@ -19,26 +20,36 @@ public Tile(int column, int row) {
     }
 
     /**
-     * The setOccupant method sets the occupant of this tile
+     * The setOccupant method sets the occupant of this tile to the piece specified
      * @param newOccupant The piece that's specified
      */
     public void setOccupant(Piece newOccupant) {
-        this.piece = newOccupant;
+        // Set the piece to occupy that tile, else assign null
+        if (newOccupant != null) {
+            this.piece = newOccupant;
+            this.occupied = true;
+        } else {
+            this.piece = null;
+            this.occupied = false;
+        }
     }
 
     /**
      * The getOccupant method determines if a tile has an occupants
      * @return The piece that is occupying this tile or null if tile has no occupant
      */
     public Piece getOccupant() {
-        return this.piece;
+        if (isOccupied()) {
+            return this.piece;
+        }
+        return null;
     }
 
     /**
      * The isOccupied method determines if a tile has a piece
      * @return True if tile has a piece, else return false
      */
     public boolean isOccupied() {
-        return false;
+        return this.occupied;
```

Comment:
You do not need a new variable to store the occupied status. You just need to return true if piece has a value, and false if it is null.